packages/feature-detector/scripts/patch-type-definition.ts

```diff
@@ -0,0 +1,106 @@
+// assumes being ran in packages/feature-detector
+
+import { get } from 'node:https'
+import { writeFile } from 'node:fs/promises'
+import { join } from 'node:path'
+
+const repositoryBaseUrl =
+  'https://raw.githubusercontent.com/microsoft/TypeScript/main/lib'
+const destinationFile = join('src', 'utils/type-definition.txt')
+
+async function writeTypeDefinition() {
+  const aggregatedContent = await fetchTypeDefinitions([
+    'lib.es2022.d.ts',
+    'lib.dom.d.ts',
+  ])
+  await writeFile(destinationFile, patchContent(aggregatedContent).join('\n'))
+}
+
+async function fetchTypeDefinitions(files: string[]) {
+  const result = []
+  for (const file of files) {
+    result.push(...(await fetchTypeDefinition(file)))
+  }
+  return result
+}
+
+async function fetchTypeDefinition(file: string): Promise<string[]> {
+  // I wish we could use node@18's fetch...
```

Comment:
Runtime, yes. But this is a build-time script. And we don't strictly need to _build_ using node16 & node14. We strictly need to _test_ the code and be sure it's running on node16 & node14.

However, to simplify CI, our test job [installs deps, builds and runs tests](https://github.com/vercel/edge-runtime/blob/main/.github/workflows/test.yml#L38-L49) on nodeX version, while we could always install & build on node 18 (and share this artifact between tests runs...), and use nodeX version for testing only.

I don't think it worth the hassle, so I rented and used builtin http utilities instead of undici.